q2_moshpit/diamond/_method.py

```diff
@@ -0,0 +1,84 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+
+from q2_types_genomics.per_sample_data import ContigSequencesDirFmt
+from q2_types_genomics.genome_data import SeedOrthologDirFmt, OrthologFileFmt
+from q2_types.feature_data import DNAFASTAFormat
+from q2_types_genomics.reference_db import DiamondDatabaseDirFmt
+
+import os
+import subprocess
+import shutil
+import tempfile
+import re
+import pandas as pd
+
+
+def eggnog_diamond_search(input_sequences: ContigSequencesDirFmt,
+                          diamond_db: DiamondDatabaseDirFmt,
+                          num_cpus: int = 1, db_in_memory: bool = False
+                          ) -> (SeedOrthologDirFmt, pd.DataFrame):
+
+    diamond_db_fp = os.path.join(str(diamond_db), 'ref_db.dmnd')
+    temp = tempfile.TemporaryDirectory()
+
+    # run analysis
+    for relpath, obj_path in input_sequences.sequences.iter_views(
+            DNAFASTAFormat):
+        sample_label = str(relpath).rsplit(r'_', 1)[0]
+
+        _diamond_search_runner(input_path=obj_path,
+                               diamond_db=diamond_db_fp,
+                               sample_label=sample_label,
+                               output_loc=temp.name,
+                               num_cpus=num_cpus,
+                               db_in_memory=db_in_memory)
+
+    result = SeedOrthologDirFmt()
+
+    for item in os.listdir(temp.name):
+        if re.match(r".*\.seed_orthologs", item):
+            shutil.copy(os.path.join(temp.name, item), result.path)
+
+    ft = extract_ft_from_seed_orthologs(result)
+
+    return (result, ft)
+
+
+def extract_ft_from_seed_orthologs(seed_orthologs: SeedOrthologDirFmt
+                                   ) -> pd.DataFrame:
+
+    per_sample_counts = []
+
+    for sample_path, obj in seed_orthologs.seed_orthologs.iter_views(
+            OrthologFileFmt):
+        sample_name = str(sample_path).rsplit(r"\.", 2)[0]
```

Comment:
##TODO Wire directly to Ortholog Directory Format path specification?

Reply:
That won't really help, but you could add a property on `OrthologFileFmt` which would be like `sample_id` which could do this logic.

Then it would just be `obj.sample_id` wherever you needed it.

Reply:
```python
@property
def sample_id(self):
    return str(self).rsplit(r'\.', 2)[0]

```

Also, why is the delimiter in the string the literal `\.`? usually that's like an underscore or something